consensus/cometbft/service/block_delay.go

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package cometbft
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/berachain/beacon-kit/primitives/encoding/json"
+)
+
+// maxDelayBetweenBlocks is the maximum delay between two consequtive blocks.
+// If the last block time minus the previous block time is greater than
+// maxDelayBetweenBlocks, then we reset `FinalizeBlockResponse.NextBlockDelay`
+// to default.
+//
+// This is needed because the network may stall for a long time and we don't
+// want to rush in new blocks as the network resumes its operation.
+const maxDelayBetweenBlocks = 30 * time.Minute
```

Comment:
What do people think is a good default here?

Reply:
@berachain/core originally I was leaning towards 2 hours, some kind of incident resolution time. But it probably does not make sense to go that large. 30 mins seems all right.